aws/resource_aws_lex_bot.go

```diff
@@ -0,0 +1,484 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/lexmodelbuildingservice"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+// Many of the Lex resources require complex nested objects. Terraform maps only support simple key
+// value pairs and not complex or mixed types. That is why these resources are defined using the
+// schema.TypeList and a max of 1 item instead of the schema.TypeMap.
+
+// Convert a slice of items to a map[string]interface{}
+// Expects input as a single item slice.
+// Required because we use TypeList instead of TypeMap due to TypeMap not supporting nested and mixed complex values.
+func expandLexObject(v interface{}) map[string]interface{} {
```

Comment:
The same approach taken in the `aws_lex_slot_type` resource to consolidate these generic expand functions should be taken for this resource.